scim2-sdk-common/src/main/java/com/unboundid/scim2/common/utils/Parser.java

```diff
@@ -322,6 +368,10 @@ private static String readPathToken(final StringReader reader)
       {
         b.append((char)c);
       }
+      else if ((c == ';') && allowSemicolonsInAttributeNames())
```

Comment:
Instead of an `allowSemicolonsInAttributeNames()` method, I'd make an `allowedExtendedCharacters()` convenience method like this:

```java
private static Set<Character> allowedExtendedCharacters()
{
  Parser.threadLocalOptions.get().getExtendedAttributeNameCharacters();
}
```

and then change the if condition to something like

```java
allowedExtendedCharacters().contains((char) c)
```

Otherwise, the ParserOptions mechanism will _only_ work if the caller configures it to allow the semicolon character, but it won't work for any other set of characters.

I think you could also get rid of this else-if block and just add the condition to the if block right above this? I.e.,

```java
if (c == '-' || c == '_' || c == '$' || Character.isLetterOrDigit(c) || allowedExtendedCharacters().contains((char) c))
{
...
```